test/e2e/specs/published-content/likes__comment.ts

```diff
@@ -64,22 +64,26 @@ describe( 'Likes: Comment', function () {
 			DataHelper.getRandomPhrase()
 		);
 
-		// For AT sites the API will respond with a HTTP 404
-		// unless time is given for the comment to "settle" in place.
-		// It could be argued that adding this arbitrary delay is
-		// "more representative" of users.
-		// @see: https://github.com/Automattic/wp-calypso/issues/75952
-		if ( envVariables.TEST_ON_ATOMIC ) {
-			await page.waitForTimeout( 5 * 1000 );
+		// The comment takes some time to settle. If we request the like
+		// immediately we might be getting the `unknown_comment` error. Let's do
+		// a few retries to make sure the like is getting through.
+		const likeRetryCount = 10;
```

Comment:
**comment:non-blocking** - I have a WIP branch locally to try to stabilize this test as well, and I tried to extract the retry mechanism to `ElementHelper` and ran into difficulties generalizing the parameter definition and forgot to ask @dpasque whether generics would help with the scenario.

What I had was the following:
```typescript
// In the test itself
		if ( envVariables.TEST_ON_ATOMIC ) {
			ElementHelper.retryRequest(
				restAPIClient.getComment(
					testAccount.credentials.testSites?.primary.id as number,
					commentToBeLiked.ID
				)
			);
		}
```

```typescript
// in ElementHelper
export async function retryRequest< T, P >(
	func: () => Promise< void >,
	parameters: [ P ],
	{ timeout = 5000 }: { timeout?: number } = {}
): Promise< void > {
	for ( let retries = 3; retries > 0; retries -= 1 ) {
		try {
			return await func( ...parameters );
		} catch ( err ) {
			// Throw the error if final retry failed.
			if ( retries === 1 ) {
				throw err;
			} else {
				await new Promise( ( resolve ) => setTimeout( resolve, timeout ) );
			}
		}
	}
	return;
}
```

Reply:
I'd even say that this should generally not be encouraged, so I'd actually avoid creating a helper for it. We need this here because of a faulty app logic which, ultimately, should be addressed somewhere else.